### PR TITLE
Refactor HiddenColumns to be compatible with upcoming IndexMappers

### PR DESCRIPTION
**Pull request by swistach**
### Context
As in title and in description of #6178. Here are necessary changes to adjust plugin to upcoming IndexMappers.

### How has this been tested?
- hide columns
- change columns order using eg.: ManualColumnMove
- insert/remove some columns
- everything should work consistently

### Types of changes
- [x] A non-breaking change

### Related issue(s):
1. #6178

### CODE REVIEW

File: src/plugins/hiddenColumns/contextMenuItem/showColumn.js [hunk 1]
```diff
@@ -2,112 +2,83 @@ import { rangeEach } from '../../../helpers/number';
...
+      hiddenBetween.length = 0;
+      hiddenAfter.length = 0;
 
-      const { from, to } = this.getSelectedRangeLast();
-      const start = Math.min(from.col, to.col);
-      const end = Math.max(from.col, to.col);
-      let hiddenInSelection = false;
+      let shouldBeHidden = true;
+      let [, start, , end] = this.getSelectedLast();
+      [start, end] = [Math.min(start, end), Math.max(start, end)];
```
Comment (wojciechczerniak) [inline src/plugins/hiddenColumns/contextMenuItem/showColumn.js]: Isn't that an unnecessary array allocation?
```suggestion
      start = Math.min(start, end);
```
```suggestion
      end = Math.max(start, end);
```

File: src/translations/indexMapper.js [hunk 2]
```diff
@@ -57,7 +57,7 @@ class IndexMapper {
   getPhysicalIndex(visualIndex) {
     const visibleIndexes = this.getNotSkippedIndexes();
     const numberOfVisibleIndexes = visibleIndexes.length;
-    let physicalIndex = null;
+    let physicalIndex = visualIndex;
```
Comment (wojciechczerniak) [inline src/translations/indexMapper.js]: This might be a breaking change. Just a note to take a look at it with @wszymanski on Monday

Comment (wszymanski) [reply src/translations/indexMapper.js]: Ye, we discussed that earlier. It will be removed.

**Comment by wszymanski** [comment]
Continued within https://github.com/handsontable/handsontable/pull/6547.

**Pull request closed without merging**